src/Sentry/ViewHierarchy.cs

```diff
@@ -0,0 +1,57 @@
+using Sentry.Extensibility;
+using Sentry.Internal.Extensions;
+
+namespace Sentry
+{
+    public class ViewHierarchy : IJsonSerializable
+    {
+        public string RenderingSystem { get; set; } = string.Empty;
+        public List<IJsonSerializable>? Children { get; set; }
+
+        public void WriteTo(Utf8JsonWriter writer, IDiagnosticLogger? logger)
+        {
+            writer.WriteStartObject();
+
+            writer.WriteStringIfNotWhiteSpace("rendering_system", RenderingSystem);
+
+            if (Children is {} children)
+            {
+                writer.WriteStartArray("windows");
+                foreach (var child in children)
+                {
+                    child.WriteTo(writer, logger);
+                }
+                writer.WriteEndArray();
+            }
+
+            writer.WriteEndObject();
+        }
+    }
+
+    public class ViewHierarchyNode : IJsonSerializable
+    {
+        public string Type { get; set; } = string.Empty;
+        public string Identifier { get; set; } = string.Empty;
+        public List<IJsonSerializable>? Children { get; set; }
+
+        public virtual void WriteTo(Utf8JsonWriter writer, IDiagnosticLogger? logger)
```

Comment:
Perhaps a better way we could do this would be to not make _this_ method virtual, but instead create a new method that is just for serializing the extra properties.  Something like `protected virtual WritePropertiesTo ...`.  Then the inheriting class wouldn't have to worry about the base properties or children.

Reply:
(Or `abstract`, if we decide the class should be abstract)